src/Host/Broker/Impl/ApiErrorResult.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.R.Host.Protocol;
+
+namespace Microsoft.R.Host.Broker {
+    internal sealed class ApiErrorResult : ObjectResult {
+        public ApiErrorResult(HttpResponse response, BrokerApiError error) : base(error) {
+            StatusCode = StatusCodes.Status500InternalServerError;
```

Comment:
Here's a decent reference that we can use as a guideline for our own errors.

https://msdn.microsoft.com/en-us/library/azure/dd179357.aspx?f=255&MSPPError=-2147217396